pkg/diagnostics/service_monitoring.go

```diff
@@ -280,3 +255,13 @@ func (s *serviceMetrics) ActorDeactivationFailed(actorType, reason string) {
 			s.actorDeactivationFailedTotal.M(1))
 	}
 }
+
+// ReportCurrentPendingLocks records the current pending actor locks.
+func (s *serviceMetrics) ReportCurrentPendingLocks(actorType, actorID string, pendingLocks int32) {
```

Comment:
I'm not sure why a user would want to see this metric of pending locks.

@amanbha

Reply:
I talked with @amanbha about turn-based concurrency related metrics. 

SF has three metrics for concurrency. I added the first one for now. 
https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-actors-diagnostics#concurrency-events-and-performance-counters

Reply:
@amanbha can elaborate on the real debugging scenario.

Reply:
This can tell us the "queue" depth which might grow in scenarios such as a blocked or long actor method call.